### PR TITLE
cephadm: fix escaping/quoting of stderr-prefix arg for ceph daemons

### PR DESCRIPTION
**Pull request by liewegas**
cephadm: fix escaping/quoting of stderr-prefix arg for ceph daemons

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
https://docs.python.org/3/library/shlex.html#shlex.quote ?

**Review by sebastian-philipp — APPROVED** [review]

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -2039,7 +2040,7 @@ def get_daemon_args(ctx, fsid, daemon_type, daemon_id):
             '--setgroup', 'ceph',
             '--default-log-to-file=false',
             '--default-log-to-stderr=true',
-            '--default-log-stderr-prefix="debug "',
+            '--default-log-stderr-prefix=debug ',
```
Comment (mgfritch) [inline src/cephadm/cephadm:2043]: (nit) maintain the quoting around the hanging space char?

**Review by mgfritch — APPROVED** [review]

File: src/cephadm/cephadm (hunk 1 above)
Comment (liewegas) [reply src/cephadm/cephadm:2043]: nope, those quotes need to go

**Comment by votdev** [comment]
This PR has introduced a regression, custom containers (type='container') can not be deployed anymore.
Example of a custom service spec:
```
service_type: container
service_id: prometheus-webhook-snmp-1
placement:
  hosts:
    - master
image: registry.suse.com/ses/7/prometheus-webhook-snmp:latest
args:
    - "--publish 9099:9099"
envs:
    - ARGS="--debug --snmp-host=192.168.121.1"
    - RUN_ARGS="--metrics"
```
The `shlex.quote` is doing too much. E.g. the additional podman command line arg `--publish 9099:9099` will be converted to `"--publish 9099:9099"` which leads to an podman error `Error: unknown flag: --publish 9099:9099`.

**Pull request merged**